### PR TITLE
New app icons

### PR DESCRIPTION
**Pull request by tht7**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [ ] This PR addresses one or more open issues that were assigned to me: ummmm sorry I needed something fun >.< I promise to be more productive tomorrow 
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
- added some alternative icons and a way to choose from them
- 
## Screenshots and Videos

https://github.com/mormaer/Mlem/assets/6067331/64a807d3-3906-4c4a-bc45-b3fcdeaf8fcd

## Additional Context
**Any additional context you'd like to add to help us review this PR**

### CODE REVIEW

**Comment by tht7** [comment]
Thank you and sorryyyy 😅 🙏

**Review by EricBAndrews — DISMISSED** [review]

**Comment by tht7** [comment]
@BrooklynMan I wanted this system to be trivial to use So I'm gonna upload a video of me adding icons:

https://drive.google.com/file/d/1SDNB6q7M0jAeOudq_yUfuLKJIQDEMKMq/view?usp=share_link

Here are the written instructions:
In the Xcode sidebar click on "Assets"
right-click the "Icons" folder
under the "iOS" section select "New iOS App Icon"
Give it a name, that name will be the one displayed in the list, if you ad in the name " By <someone>" that "<someone>" will appear as a grey subtitle in the list (because credit is important especially in open-source)
You'll then be presented with a square that says "1024x1024" on top of which you can drop your image file
(not that the image file _**must**_ be in the correct format else it won't show up on the list and you'll see a yellow warning mark next to the image)

and that's it! no code to edit! no plists to mess with

I encourage you to go wild 😝

File: Mlem/Views/Tabs/Settings/Components/Views/AlternativeIconCell.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                        Text(author)
+                            .font(.footnote)
+                            .foregroundColor(.secondary)
+                    }
+                }
+                Spacer()
+                if icon.selected {
+                    Image(systemName: "checkmark")
+                }
+            }
```
Comment (WestonHanners) [inline Mlem/Views/Tabs/Settings/Components/Views/AlternativeIconCell.swift:34]: I think we want `.accessibilityElement(children: .combine)` here.

Comment (tht7) [reply Mlem/Views/Tabs/Settings/Components/Views/AlternativeIconCell.swift:34]: Ohhh thanks, I need to learn about accessibility eventually

**Comment by BrooklynMan** [comment]
@tht7 this is the proper icon name list:

**Custom Icon Name List**

- Mlem
- Mlem Pride
- Mlem Inclusive Pride
- Mlem Trans Pride
- Mlem Wave
- Mlem Cream
- Mlemkin
- Mlemhaw
- Mlem Dark
- Mlem Dev
- Mlemmygrad
- Mlem Mono

**Comment by tht7** [comment]
Mlem Mono is in the incorrect format on drive, do you mind scaling it back to 1024/1024px

**Review by BrooklynMan — APPROVED** [review]
lovely!

**Pull request merged**